presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/ResourceEstimatesRule.java

```diff
@@ -0,0 +1,193 @@
+    }
+
+    @JsonProperty
+    public Optional<ResourceEstimateRange<Duration>> getExecutionTimeRange()
+    {
+        return executionTimeRange;
+    }
+
+    @JsonProperty
+    public Optional<ResourceEstimateRange<Duration>> getCpuTimeRange()
+    {
+        return cpuTimeRange;
+    }
+
+    @JsonProperty
+    public Optional<ResourceEstimateRange<DataSize>> getPeakMemoryRange()
+    {
+        return peakMemoryRange;
+    }
+
+    boolean match(ResourceEstimates resourceEstimates)
+    {
+        if (executionTimeRange.isPresent() &&
+                !(resourceEstimates.getExecutionTime().isPresent() && executionTimeRange.get().checkWithinRange(resourceEstimates.getExecutionTime().get()))) {
+            return false;
+        }
+
+        if (cpuTimeRange.isPresent() &&
+                !(resourceEstimates.getCpuTime().isPresent() && cpuTimeRange.get().checkWithinRange(resourceEstimates.getCpuTime().get()))) {
+            return false;
+        }
+
+        if (peakMemoryRange.isPresent() &&
+                !(resourceEstimates.getPeakMemory().isPresent() && peakMemoryRange.get().checkWithinRange(resourceEstimates.getPeakMemory().get()))) {
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    public String toString()
+    {
+        final StringBuilder sb = new StringBuilder("ResourceEstimatesRule{");
+        sb.append("executionTimeRange=").append(executionTimeRange);
+        sb.append(", cpuTimeRange=").append(cpuTimeRange);
+        sb.append(", peakMemoryRange=").append(peakMemoryRange);
+        sb.append('}');
+        return sb.toString();
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(executionTimeRange, cpuTimeRange, peakMemoryRange);
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        ResourceEstimatesRule other = (ResourceEstimatesRule) o;
+
+        return Objects.equals(this.executionTimeRange, other.executionTimeRange) &&
+                Objects.equals(this.cpuTimeRange, other.cpuTimeRange) &&
+                Objects.equals(this.peakMemoryRange, other.peakMemoryRange);
+    }
+
+    public static class ResourceEstimateRange<T extends Comparable<T>>
+    {
+        private final Optional<T> min;
+        private final Optional<T> max;
+
+        @JsonCreator
+        public ResourceEstimateRange(
+                @JsonProperty("min") Optional<T> min,
+                @JsonProperty("max") Optional<T> max)
+        {
+            this.min = requireNonNull(min, "min is null");
+            this.max = requireNonNull(max, "max is null");
+        }
+
+        public boolean checkWithinRange(T value)
+        {
+            return (!min.isPresent() || min.get().compareTo(value) <= 0) &&
```

Comment:
Why not use `com.google.common.collect.Range`?

Reply:
That's a good idea! But it doesn't support to be serialized to JSON right?